packages/core/src/lib/annotationExporter.js

```diff
@@ -6,39 +6,47 @@ const _ = require('lodash');
 const mp = require('./markdown_parser');
 const logger = require('./log');
 
-const annotations_exporter = function (pl) {
+const annotationExporter = function (pl) {
   const paths = pl.config.paths;
-  let oldAnnotations;
 
   /**
    * Parses JS annotations.
    * @returns array of comments that used to be wrapped in raw JS
    */
-  function parseAnnotationsJS() {
+  function parseAnnotationsJSON() {
+    const jsonPath = path.resolve(paths.source.annotations, 'annotations.json');
+    let annotations;
+
     //attempt to read the file
     try {
-      oldAnnotations = fs.readFileSync(
-        path.resolve(paths.source.annotations, 'annotations.js'),
-        'utf8'
-      );
+      if (fs.pathExistsSync(jsonPath)) {
+        //read the new file
+        annotations = fs.readFileSync(jsonPath, 'utf8');
+      } else {
+        //read the old file
+        const jsPath = path.resolve(paths.source.annotations, 'annotations.js');
+
+        annotations = fs
+          .readFileSync(jsPath, 'utf8')
+          .replace(/^\s*var comments ?= ?/, '')
+          .replace(/};\s*$/, '}');
+
+        logger.info(
+          `Please convert ${jsPath} to JSON and rename it annotations.json.`
+        );
```

Comment:
I'm a bit on the fence with that. We should let the user decide if he wants to use JS or JSON, even in the future.
Furthermore, we could check if it is possible to `require` that `.js` or `.json` file so that we do not have to strip comments and anything else in that file. Maybe we can change the way it needs to be written to a `module.exports` format.

Reply:
@JosefBredereck, actually I've realized that we're stripping that `var comments` part in the first place, to add it again on exporting the data: https://github.com/pattern-lab/patternlab-node/blob/41b4ec15243ba8b1e56bb1efce7d39dfe4c4e150/packages/core/src/lib/exportData.js#L71

So another solution might be to skip that whole transition to a pure JSON file and only provide JS (the existing format), but remove the JSON file possibility, as we're even already proving another format with the markdown notation as well.

Reply:
No way 😄. In that case, throw away those nasty JSON files 🗑️

Reply:
okay, sounds reasonable to me. I've put this ticket on draft again and will proceed by removing these parts as well as the nasty existing transitions in the codebase, as well as correct the documentation page.

Reply:
The whole file you mentioned is a solution, but if we would rework the codebase with today's standards, we would maybe not do it that way.

Reply:
so does it mean that you want to come up with (or at least describe) a solution ?

Reply:
Not for now; it would require a more significant rewrite.
This line shows that we already work with modules somewhere.

https://github.com/pattern-lab/patternlab-node/blob/41b4ec15243ba8b1e56bb1efce7d39dfe4c4e150/packages/core/src/lib/exportData.js#L88

But then we also need to adjust the place where we load those annotations again.

What makes me curious. Annotations always have a comments property and no other. Is it possible that they have different properties? If not, this would be another change I would like to implement somehow.

https://github.com/pattern-lab/patternlab-node/blob/41b4ec15243ba8b1e56bb1efce7d39dfe4c4e150/packages/core/src/lib/exportData.js#L72

Reply:
> No way 😄. In that case, throw away those nasty JSON files 🗑️

On the other hand, besides the strange way to include the JSON data into the website after to a transformation, the positive way of using JSON (even only) would be that this adapts the way other data is being maintained in the pattern lab system (like pattern data).

Reply:
That is correct. Nodejs is so fast in transforming the data that we could use the approach that is already in this PR. In the end, it makes no difference.